File: app/src/main/java/com/datarockets/mnchkn/activities/onboard/OnboardInteractorImpl.java
package com.datarockets.mnchkn.activities.onboard;

import android.content.Context;

import com.datarockets.mnchkn.store.SettingsServiceImpl;

public class OnboardInteractorImpl implements OnboardInteractor {

    private SettingsServiceImpl mSettingsService;

    public OnboardInteractorImpl(Context context) {
        mSettingsService = new SettingsServiceImpl(context);
    }

    @Override
    public void isUserSeenOnboarding(OnFinishedChecking listener) {
        listener.shouldShowOnboarding(!mSettingsService.checkIsUserSeenOnboarding());
    }

    @Override
    public void setOnboardingSeen() {
        mSettingsService.setOnboardingSeen();
    }

}

File: app/src/main/kotlin/com/datarockets/mnchkn/activities/onboard/OnboardInteractorImpl.kt
package com.datarockets.mnchkn.activities.onboard

import android.content.Context

import com.datarockets.mnchkn.MunchkinApplication
import com.datarockets.mnchkn.store.SettingsService

import javax.inject.Inject

class OnboardInteractorImpl(context: Context) : OnboardInteractor {

    @Inject
    lateinit var mSettingsService: SettingsService

    init {
        MunchkinApplication.get(context).appComponent.inject(this)
    }

    override fun isUserSeenOnboarding(listener: OnboardInteractor.OnFinishedChecking) {
        listener.shouldShowOnboarding(!mSettingsService.checkIsUserSeenOnboarding())
    }

    override fun setOnboardingSeen() {
        mSettingsService.setOnboardingSeen()
    }

}
